check.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+#
+# LICENSE: Public Domain
+
+exitcode=0
+
+echo -e "\nCHECK that all translator have a translatorID..."
+HAVEID=$(grep -c '"translatorID"' *.js | awk -F: '$2!="1" {print $1,"found",$2,"translatorID"}')
+if [[ -n "$HAVEID" ]];then
+  echo "Error: no translatorID or multiple translatorIDs"
+  echo "$HAVEID"
+  exitcode=1
+fi
+echo "...DONE"
+
+
+echo -e "\nCHECK that all translatorIDs are unique..."
+USEDID=$(grep '"translatorID"' *.js | cut -d: -f3)
+DELETEDID=$(tail -n +3 deleted.txt | cut -d# -f1)
+NODUPLICATE=$(echo -e "${USEDID}\n${DELETEDID}" | sed 's/[", ]//g' | sort | uniq -d)
+if [[ -n "$NODUPLICATE" ]];then
+  echo "Error: duplicate translatorID found"
+  echo "$NODUPLICATE"
+  exitcode=1
+fi
+echo "...DONE"
+
+
+echo -e "\nCHECK on all translators (file mode 644, file ending, deprecated JS for each)..."
+for file in *.js; do
+  filemode=$(stat -c "%a" "$file")
+  if [[ "$filemode" != "644" ]];then
+    echo "Error: file mode is $filemode =/= 644 in $file"
+    exitcode=1
+  fi
+  dosfilending=$(dos2unix < "$file" | cmp -s - "$file")
+  if [[ -n "$dosfilending" ]];then
+    echo "Error: wrong file ending in $file"
+    exitcode=1
+  fi
+  foreach=$(grep -E 'for each *\(' *.js)
+  if [[ -n "$foreach" ]];then
+    echo "Error: Deprecated JavaScript for each is used in $file"
+    exitcode=1
+  fi
+done
+echo "...DONE"
+
+
+echo -e "\nTEST output \$TRAVIS_COMMIT_RANGE  $TRAVIS_COMMIT_RANGE"
```

Comment:
This evaluates here to [43f97c986d22...caa8f7708e2b](https://github.com/zuphilip/translators/compare/43f97c986d22...caa8f7708e2b) which is just containing the last commit...?